cpp/zotaut/dump_delivered_marc_records.cc

```diff
@@ -79,19 +79,36 @@ void GetJournalEntriesFromDb(DbConnection * const db_connection, const std::stri
             const auto dois(record.getDOIs());
             csv_row += ";" + TextUtil::CSVEscape(StringUtil::Join(dois, '\n'));
 
-            for (const auto &_936_field : record.getTagRange("936")) {
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('j');
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('d');
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('e');
-                break;
+            const auto _773_field(record.findTag("773"));
+            const auto g_773_contents(_773_field->getFirstSubfieldWithCode('g'));
+            if (not g_773_contents.empty()) {
+                std::vector<std::string> subfields;
+                std::vector<std::string> filtered_dates;
+                for (const auto &field : record.getTagRange("773")) {
```

Comment:
This iteration should be improved because you call findTag() and getTagRange() twice on the same 773 field. And you should check the indicator before trying to get any subfields.